x-pack/plugins/cloud/public/plugin.tsx

```diff
@@ -147,6 +176,12 @@ export class CloudPlugin implements Plugin<CloudSetup> {
       // In the event of an unexpected error, fail *open*.
       // Cloud admin console will always perform the actual authorization checks.
       .catch(() => setLinks(true));
+
+    return {
+      CloudContextProvider: ({ children }) => (
+        <ServicesProvider chat={this.chatService}>{children}</ServicesProvider>
+      ),
```

Comment:
If the request performed in `setupChat` takes enough time to complete after the plugin starts, there is a risk here that a consumer would get a provider with an obsolete `chat` value, which could cause the chat to not be displayed even if enabled. 

This can be worked around by using an observable for `ServicesProvider.chat` instead of a plain value.

Reply:
Ah... I'll give this a try, thanks.

Reply:
I tried to do this and ran into significant trouble... Observables are not my forte.  @pgayvallet is there any chance you could help me out here?